### PR TITLE
Create instagram_pic

### PR DESCRIPTION
**Pull request by Epic-R-R**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/instagram_pic.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import requests
+from bs4 import BeautifulSoup
+import datetime
```
Comment (cclauss) [inline web_programming/instagram_pic.py:3]:
```suggestion
import requests
from bs4 import BeautifulSoup
from datetime import datetime
```
Please run `isort` on the imports.

File: web_programming/instagram_pic.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+if __name__ == "__main__":
+    url = input("Enter image url: ")
+    print("Downloading image...")
```
Comment (cclauss) [inline web_programming/instagram_pic.py:7]:
```suggestion
    print(f"Downloading image from {url} ...")
```
Use f-strings as discussed in CONTRIBUTING.md

File: web_programming/instagram_pic.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    req = requests.get(url)
+    soup = BeautifulSoup(req.content, "html.parser")
```
Comment (cclauss) [inline web_programming/instagram_pic.py:9]:
```suggestion
    soup = BeautifulSoup(requests.get(url).content, "html.parser")
```
Avoid creating variables that are only used on the next line unless:
1. The lines are already close to the max_line_length of 88 chars per line -- or --
2. The variable name clarifies something that is unclear

File: web_programming/instagram_pic.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    metaTag = soup.find_all("meta", {"property": "og:image"}) #get all meta tags
+    imgURL = metaTag[0]["content"]
```
Comment (cclauss) [inline web_programming/instagram_pic.py:11]:
```suggestion
    # The image URL is in the content field of the first meta tag with the property og:image
    image_url = soup.find("meta", {"property": "og:image"})["content"]
```

Comment (cclauss) [reply web_programming/instagram_pic.py:11]: Python variable names are in snake_case as discussed in CONTRIBUTING.md.

File: web_programming/instagram_pic.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    r = requests.get(imgURL)
+    fileName = datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S") + ".jpg"
+    with open(fileName, "wb") as fp:
+        fp.write(r.content)
```
Comment (cclauss) [inline web_programming/instagram_pic.py:15]:
```suggestion
    image_data = requests.get(imgURL).content
    file_name = f"{datetime.now():%Y-%m-%d_%H:%M:%S}.jpg"
    with open(file_name, "wb") as fp:
        fp.write(image_data)
```
Avoid single-letter variable names -- They make code look like it was written in the 1970's.  The reader of this code does not care about the response, but they do care about the image_data so focus their attention on that.

Use f-strings which are more expressive especially with complex types like datetimes.

File: web_programming/instagram_pic.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    print("Done. Image saved to disk as " + fileName)
```
Comment (cclauss) [inline web_programming/instagram_pic.py:16]:
```suggestion
    print(f"Done. Image saved to disk as {file_name}.")
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: web_programming/instagram_pic.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import requests
+from bs4 import BeautifulSoup
+import datetime
+
+if __name__ == "__main__":
+    url = input("Enter image url: ")
```
Comment (cclauss) [inline web_programming/instagram_pic.py:6]:
```suggestion
    url = input("Enter image url: ").strip()
```
As discussed in CONTRIBUTING.md.

**Comment by erdum** [comment]
Plz add type hints in your function parameters like this, def test(a: int, b: int) -> int:
Like that you have to define the types of parameters and the type of return of the function,
And plz jus install doctest pip3 install doctest, then read this tutorial on how to use doctest in your code https://www.geeksforgeeks.org/testing-in-python-using-doctest-module/, then commit changes and hope pre-commit check will be clear thanks.

**Comment by Epic-R-R** [comment]
I corrected the code and commit it, but it still gives an error in pre-commit isort field

**Comment by erdum** [comment]
Yes i saw that your flake8 and isort is failing

**Comment by erdum** [comment]
Install isort in your machine and run isort in this file locally

**Comment by erdum** [comment]
And also install flake8 to see what styling mistake you done and if you can't corrected manually then install black and run black on this file it will correct all your styling mistake and flake8 check should be clear

**Comment by Epic-R-R** [comment]
thanks, I use isort and then All checks have passed

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Nice work!  Thanks for doing this.

**Comment by erdum** [comment]
cclauss please also review my pull request

**Pull request merged**